include/net/socket.h

```diff
@@ -957,6 +769,14 @@ struct net_socket_register {
 
 #include <syscalls/socket.h>
 
+#ifdef CONFIG_NET_SOCKETS_POSIX_NAMES
+/* These prefixed with posix as they can be used without CONFIG_POSIX_API */
+#include <posix/arpa/inet.h>
```

Comment:
@rlubos - I might remove `CONFIG_NET_SOCKET_POSIX_NAMES` from `lib/posix/CMakeLists.txt` and add 
```cpp
#include <posix/fcntl.h> // fcntl(2)
#include <posix/unistd.h> // close(2)
// also maybe remove <posix/sys/socket.h> here, since there is already <net/socket_select.h>
```

What are your thoughts on that?

Reply:
Technically, a lot of the POSIX stuff is just defined directly in `net/socket.h`, but some things should probably be prefixed (even though there is a 1:1 relationship). E.g. `TCP_NODELAY`.